### DISCUSSION TITLE
How to grab page number when viewing PDF?

### DISCUSSION THREAD

**Discussion by dlanRT** [MicrosoftEdge/WebView2Feedback#3914] [Q&A]
I am trying to grab and save the page number of a PDF so that it can be saved and returned to when a user closes the WebView and returns. However at the moment when attempting to use the ExecuteScriptAsync command to run a 'getElementById' command all that gets returned is null. For any div ID I try to search. The command works just fine if run through dev tools in console, so the id is correct. I also noticed when just trying to return the 'document' all that gets returned is a JSON object that contains a hashed value and the path to the PDF file in question. Everything works fine if attempting this through a regular web page however. Is this just the behavior of the PDF Viewer?
 
To shorten my question, is there another way to grab the current page of a PDF in WebView2?

**Comment by Felix-CodingClimber** [comment]
Same problem here! Trying to interact with the pdf viewer don't work for me by using ExecuteScriptAsync... within the dev console it work's fine.

**Comment by ritesh3103** [comment]
Till date, this question is unanswered. I am having this problem with Windows App SDK 1.1.3 (Visual Studio 2022 17.2.6)

**Comment by alan008** [comment]
Need this function also.
And full control to loaded PDF via API (Change zoom, navigate to pages, set zoom mode (by width, by page) etc.
